sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Sample.java

```diff
@@ -246,6 +254,36 @@ private SampleAnyCombineFn(long limit) {
     }
   }
 
+  /** A {@link CombineFn} that combines into a single element. */
+  private static class AnyValueCombineFn<T> extends CombineFn<T, List<T>, T> {
+    private SampleAnyCombineFn internal;
+
+    private AnyValueCombineFn() {
+      internal = new SampleAnyCombineFn<>(1);
+    }
+
+    @Override
+    public List<T> createAccumulator() {
+      return internal.createAccumulator();
+    }
+
+    @Override
+    public List<T> addInput(List<T> accumulator, T input) {
+      return internal.addInput(accumulator, input);
+    }
+
+    @Override
+    public List<T> mergeAccumulators(Iterable<List<T>> accumulators) {
+      return internal.mergeAccumulators(accumulators);
+    }
+
+    @Override
+    public T extractOutput(List<T> accumulator) {
+      Iterator<T> it = internal.extractOutput(accumulator).iterator();
+      return it.hasNext() ? it.next() : null;
```

Comment:
I think this is right to return a null. 

per [1], if empty or input rows are all null, return null.

Though returning `null` in Java sometimes do causing problems (not the problem in SQL layer, but in Java world). But in BeamSQL we just did it so far.


[1]: https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#any_value

Reply:
Yes, I was aware of that. I thought that I could have used java Optional<T> or similar, but I was not sure it would be compliant with other components. Are there other functions that return optional or nullable objects in the Java SDK?, I could give them a look.

Reply:
BTW, I created an [issue](https://issues.apache.org/jira/browse/BEAM-9864) for OVER/window clauses as you suggested.